packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -980,27 +972,24 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
     final Matrix4 transform = target.getTransformTo(this);
     Rect targetRect = MatrixUtils.transformRect(transform, rect);
 
-    final double mainAxisExtent = switch (axisDirectionToAxis(axisDirection)) {
-      Axis.horizontal => viewportDimension.width,
-      Axis.vertical => viewportDimension.height,
+    final double mainAxisExtentDifference = switch (axis) {
+      Axis.horizontal => viewportDimension.width - rectLocal.width,
+      Axis.vertical   => viewportDimension.height - rectLocal.height,
     };
 
-    final double targetOffset = leadingScrollOffset - (mainAxisExtent - targetMainAxisExtent) * alignment;
+    final double targetOffset = leadingScrollOffset - mainAxisExtentDifference * alignment;
```

Comment:
`mainAxisExtent` and `targetMainAxisExtent` are only used to set the value of `targetOffset`, so we can consolidate these into a single variable by performing the subtraction beforehand.